From 34fb311eabdedc24000baa1351953f56e787f949 Mon Sep 17 00:00:00 2001
From: barracuda156 <barracuda@macos-powerpc.org>
Date: Mon, 31 Aug 2026 02:34:19 +0000
Subject: [PATCH 09/52] port: cpu.cc-support-powerpc-darwin

[Re-anchored onto 12.4 from the 9.4 series; hunks applied without modification.]
---
 v8/src/base/cpu.cc | 28 +++++++++++++++++++++++++++-
 1 file changed, 27 insertions(+), 1 deletion(-)

diff --git a/v8/src/base/cpu.cc b/v8/src/base/cpu.cc
index 1b1c5c0..1008206 100644
--- a/v8/src/base/cpu.cc
+++ b/v8/src/base/cpu.cc
@@ -38,6 +38,15 @@
 #if V8_OS_DARWIN
 #include <sys/sysctl.h>  // sysctlbyname
 #endif
+#if V8_OS_MACOSX
+#include <sys/sysctl.h>  // sysctlbyname
+#if V8_HOST_ARCH_PPC || V8_HOST_ARCH_PPC64
+#include <mach/mach.h>
+#include <mach/mach_host.h>
+#include <mach/host_info.h>
+#include <mach/machine.h>
+#endif
+#endif
 #if V8_OS_POSIX
 #include <unistd.h>  // sysconf()
 #endif
@@ -908,7 +917,24 @@ CPU::CPU()
       part_ = kPPCPower5;
       break;
   }
-#endif  // V8_OS_AIX
+
+#elif V8_OS_MACOSX
+
+struct host_basic_info host_basic_info;
+  switch(host_basic_info.cpu_subtype) {
+    case CPU_SUBTYPE_POWERPC_970:
+    default:
+      part_ = kPPCG5;
+      break;
+    case CPU_SUBTYPE_POWERPC_7450:
+      part_ = kPPCG4;
+      break;
+    case CPU_SUBTYPE_POWERPC_7400:
+      part_ = kPPCG4;
+      break;
+   }
+
+#endif  // V8_OS_MACOSX
 #endif  // !USE_SIMULATOR
 
 #elif V8_HOST_ARCH_RISCV64
-- 
2.43.0

